### DISCUSSION TITLE
Is it possible to print matching file and line number but omit the match?

### DISCUSSION THREAD

**Discussion by frankitox** [BurntSushi/ripgrep#2746] [Q&A] [answered]
So, I want to print something like:

```bash
resources/public/externaljs/video.js:21914
resources/public/externaljs/video.js:21915
resources/public/externaljs/video.js:21916
```

The closest thing I managed to write is:

```bash
rg --no-column --line-number --no-heading --only-matching --replace= attrs
resources/public/externaljs/video.js:21914:
resources/public/externaljs/video.js:21915:
resources/public/externaljs/video.js:21916:
```

I'm happy with this as it is, but there's any flag I could use to erase that last semicolon?

**Comment by BurntSushi** [comment, marked as answer]
Nope. I'd probably just throw a `sed 's/:$//'` at the end of it and call it good enough.

If you're reading this data into some other program, you might also be interested in the `--json` flag.

**Comment by frankitox** [comment]
Gotcha, thanks!!

**Comment by kranurag7** [comment]
Hey there, I had the same question and I have been searching for the answer in discussion for few weeks now from time to time. 

The reason this is important to me is because some editor primarily helix allows you to open files by number and columns extended to it. 
```bash
hx boilerplate.py:19:12
```
I tried getting similar answers with grep as well and found the following result. 
https://stackoverflow.com/questions/8105685/show-filename-and-line-number-in-grep-output 

The above doesn't solve my problem either, given they also append the query next to the search line number. 
```bash
$ grep -Horn python
boilerplate.py:1:python
boilerplate.py.txt:1:python
```
I was looking for the search results filtered by line only. 

The solution suggested above works good. 
```bash
$ rg --no-column --line-number --no-heading --only-matching --replace= python | sed 's/:$//'
testdata/with_config/pass.py:1
testdata/with_config/fail.py:1
```
And now I can open all the match using the following command. 
```bash
rg --no-column --line-number --no-heading --only-matching --replace= python | sed 's/:$//' | xargs hx 
```
But this is too long. I wonder it there's a more simpler way to achieve the same. As of now, I trimmed it down to something like using short flags. 
```bash
$ rg -r= -o -n --no-heading --no-column python
```

**Reply by BurntSushi** [reply]
Yes, the OP mentioned `--replace=` as well. You still end up with the trailing `:`.

ripgrep doesn't expose a way to eliminate that trailing colon and likely never will.

> The reason this is important to me is because some editor primarily helix allows you to open files by number and columns extended to it.

This part I don't understand. Why do you need to remove the matching part of the line in order to achieve this goal?

**Reply by kranurag7** [reply]
```bash
# in a directory I've a file called hello.md
$ cat hello.md
ripgrep
is
awesome
and
i
use
it
everyday.
$ rg awesome
hello.md
3:awesome

# this is the point where I want the output to look something like `hello.md:3` meaning that `awesome` is located in hello.md file at 3rd row.
# If I get the output as `hello.md:3` then I can use my editor `helix` to open the file and the cursor will be at 3rd row if I invoke the command `hx hello.md:3` 

# now the second case
$ rg some --column
hello.md
3:4:awesome

# In above case, ripgrep shows me that the word `some` is located at 3rd row and fourth column in a file hello.md 
# here in this case, I want the output to be like `hello.md:3:4` so that I can use it with my editor by invoking the `hx hello.md:3:4` command. 
```

The solution mentioned above works and I'll put it behind an alias but just thinking if we can get it using a more simpler or using an acronym that's easy to recall and type. 
Thank you very much. Please ask if you need any more additional inputs from my side.

**Reply by BurntSushi** [reply]
Okay but that doesn't really answer my question. I understand the output format you want. That's not what I'm asking about. What I'm asking is why `path:3:4:the matched line` can't be used in the context of Helix? If I had to guess, it sounds like Helix expects the `path:3:4` format and probably chokes if `:the matched line` is present? Is that right?

If so, then yeah you don't really have a choice here. Just use the `sed` command. And maybe put it in a wrapper script or alias.

**Reply by kranurag7** [reply]
Yes, that is correct. It created a new file. 

```bash
$ rg awesome --no-heading | xargs hx
# save and quit helix editor
# I get a new file named hello.md:awesome
```
Helix doesn't recognize this syntax and creates a new file. 

Thank you very much for the answer, I'll put the syntax with `sed` text-processing included behind an alias.

**Reply by frankitox** [reply]
Glad the code is of any help @kranurag7 ! I'm using it to feed a fzf command that's called from my editor (neovim). Luckily the `:` at the end doesn't break anything and it's merely esthetic.
Totally unrelated, but I would recommend using long flags if possible, it is always easier to remember/understand later.
